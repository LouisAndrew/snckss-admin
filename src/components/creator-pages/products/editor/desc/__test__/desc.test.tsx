import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import Desc from '..'

describe('Description input element', () => {
        let temp: string = ''

        const mockHeaderText: string = 'Mock Header'
        const mockPlaceholderText: string = 'Mock placeholder'
        const mockDesc: string = ''
        const mockHandleChange = jest.fn(
                (event: React.ChangeEvent<HTMLInputElement>) => {
                        temp = event.target.value
                }
        )

        afterEach(cleanup)
        const el: React.ReactElement = (
                <Desc
                        headerText={mockHeaderText}
                        placeholderText={mockPlaceholderText}
                        desc={mockDesc}
                        handleChange={mockHandleChange}
                />
        )

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        it('should render the placeholder text correctly', () => {
                const { queryByPlaceholderText } = render(el)
                const element: HTMLElement | null = queryByPlaceholderText(
                        mockPlaceholderText
                )
                if (element) {
                        expect(element).toBeInTheDocument()
                } else {
                        fail()
                }
        })

        it('should render the label text correctly', () => {
                const { queryByLabelText } = render(el)
                const element: HTMLElement | null = queryByLabelText(
                        mockHeaderText
                )
                if (element) {
                        expect(element).toBeInTheDocument()
                } else {
                        fail()
                }
        })

        it('should respond to the input function correctly', () => {
                const mockUserInput: string = 'user input'

                const { getByLabelText } = render(el)
                const element: HTMLElement = getByLabelText(mockHeaderText)
                fireEvent.change(element, { target: { value: mockUserInput } })

                expect(temp).toBe(mockUserInput)
        })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
