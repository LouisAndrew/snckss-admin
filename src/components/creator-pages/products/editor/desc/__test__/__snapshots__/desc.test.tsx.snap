// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description input element matches snapshot 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    htmlFor="desc"
  >
    Mock Header
  </label>
  <textarea
    className="form-control"
    id="desc"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            SyntheticEvent {
              "_dispatchInstances": null,
              "_dispatchListeners": null,
              "_targetInst": null,
              "bubbles": null,
              "cancelable": null,
              "currentTarget": [Function],
              "defaultPrevented": null,
              "dispatchConfig": null,
              "eventPhase": null,
              "isDefaultPrevented": [Function],
              "isPropagationStopped": [Function],
              "isTrusted": null,
              "nativeEvent": null,
              "target": null,
              "timeStamp": [Function],
              "type": null,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    placeholder="Mock placeholder"
    value=""
  />
</div>
`;
